platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/annotations/BubbleLayout.java

```diff
@@ -0,0 +1,233 @@
+package com.mapbox.mapboxsdk.annotations;
+
+import android.content.Context;
+import android.content.res.TypedArray;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.RectF;
+import android.util.AttributeSet;
+import android.util.DisplayMetrics;
+import android.widget.LinearLayout;
+
+import com.mapbox.mapboxsdk.R;
+
+/**
+ * Bubble View for Android with custom stroke width and color, arrow size, position and direction.
+ */
+public class BubbleLayout extends LinearLayout {
+
+  public static final float DEFAULT_STROKE_WIDTH = -1;
+
+  private ArrowDirection mArrowDirection;
+  private Bubble mBubble;
+
+  private float mArrowWidth;
+  private float mCornersRadius;
+  private float mArrowHeight;
+  private float mArrowPosition;
+  private int mBubbleColor;
+  private float mStrokeWidth;
+  private int mStrokeColor;
```

Comment:
same remark for this class as in https://github.com/mapbox/mapbox-gl-native/pull/8080/files#r101518527